cockatrice/src/chatview.cpp

```diff
@@ -161,72 +161,115 @@ void ChatView::appendMessage(QString message, QString sender, UserLevelFlags use
     }
     cursor.setCharFormat(messageFormat);
     
-    int index = -1, bracketFirstIndex = -1, mentionFirstIndex = -1, urlFirstIndex = -1;
+    int index = -1, bracketFirstIndex = -1, mentionFirstIndex = -1, urlFirstIndex = -1, highlightWordFirstIndex = -1;
     bool mentionEnabled = settingsCache->getChatMention();
     const QRegExp urlStarter = QRegExp("https?://|\\bwww\\.");
     const QRegExp phraseEnder = QRegExp("\\s");
     const QRegExp notALetterOrNumber = QRegExp("[^a-zA-Z0-9]");
+    const QStringList highlightedWords = settingsCache->getHighlightWords();
     
     while (message.size())
     {
-        // search for the first [ or @
         bracketFirstIndex = message.indexOf('[');
         mentionFirstIndex = message.indexOf('@');
         urlFirstIndex = message.indexOf(urlStarter);
+        highlightWordFirstIndex = -1;
+
+        foreach (QString word, message.simplified().split(" "))
```

Comment:
You need to check if there are words in the list first, that way you can avoid addition computation of splitting the message for each pass. 

You also split the same message over and over for each pass. This adds up when you have a `message` with a few mentions, cards or matching words. As the message doesnt change during processing, maybe you could split it once and work with that instead.

Why are you removing internal chars with simplified() why not use trimmed() instead?